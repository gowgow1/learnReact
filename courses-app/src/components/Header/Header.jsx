import Logo from './components/Logo';
import Button from '../common/Button';

import './index.css';

const Header = () => (
	<div className='header'>
		<Logo />
		<div className='header-wrap'>
			<div className='header-text'>Dave</div>
			<Button text='logout' onClick={() => {}} />
		</div>
	</div>
);

export default Header;
