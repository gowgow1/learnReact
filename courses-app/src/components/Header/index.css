.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 5px 20px;
  margin-bottom: 20px;
  border: 2px solid lightsalmon;
}

.header-text {
  font-size: 20px;
  font-weight: 600;
}
.header-wrap {
  display: flex;
  margin-left: 40px;
  justify-content: center;
  align-items: center;
  column-gap:  40px;
}

.header .button {
  padding: 5px 20px;
  text-transform: uppercase;
}