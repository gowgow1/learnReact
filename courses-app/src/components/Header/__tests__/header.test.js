import * as reactRedux from 'react-redux';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import Header from '../Header';

const mockedState = {
	isAuth: true,
	name: 'Alex',
	token: 'f1f2f3f4',
	role: 'user',
	email: 'alex@mail.ru',
};

const mockedStore = {
	getState: () => mockedState,
	subscribe: jest.fn(),
	dispatch: jest.fn(),
};

jest.spyOn(reactRedux, 'useDispatch');

jest.mock('react-redux', () => ({
	...jest.requireActual('react-redux'),
	useSelector: jest.fn(),
}));

beforeEach(() => {
	reactRedux.useSelector.mockReturnValue(mockedState);
});

afterEach(() => {
	reactRedux.useSelector.mockClear();
});

describe('Testing: <Header/>', () => {
	it('rendered without crushing', () => {
		const { container } = render(
			<Provider store={mockedStore}>
				<Header />
			</Provider>
		);
		expect(container).toMatchSnapshot();
	});
	it('should get name from store', async () => {
		const { getByText } = render(
			<Provider store={mockedStore}>
				<Header />
			</Provider>
		);
		expect(getByText('Alex')).toBeTruthy();
	});

	it('should be with alt', () => {
		const { queryByAltText } = render(
			<Provider store={mockedStore}>
				<Header />
			</Provider>
		);
		expect(queryByAltText('logo')).toBeTruthy();
	});
});
