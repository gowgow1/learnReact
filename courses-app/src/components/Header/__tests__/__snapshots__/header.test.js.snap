// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: <Header/> rendered without crushing 1`] = `
<div>
  <div
    class="header"
  >
    <img
      alt="logo"
      class="logo"
      src="react.svg"
    />
    <div
      class="header-wrap"
    >
      <div
        class="header-text"
      >
        Alex
      </div>
      <button
        class="button"
      >
        logout
      </button>
    </div>
  </div>
</div>
`;
